model/headers/headers.go

```diff
@@ -135,9 +144,22 @@ func NewHeader(reader io.Reader, readerPrivateKey [chacha20poly1305.KeySize]byte
 	for i := uint32(0); i < header.HeaderPacketCount; i++ {
 		headerPacket, err := NewHeaderPacket(reader, readerPrivateKey)
 		if err != nil {
-			return nil, err
+			switch err := err.(type) {
+			case *HeaderReaderError:
+				// do nothing
+				// we carry on and try the next header package
+			default:
+				// for any other error we return it
+				return nil, err
+			}
+		}
+		if headerPacket != nil {
+			header.HeaderPackets = append(header.HeaderPackets, *headerPacket)
```

Comment:
Will this ever be case, i.e. that the headerPacket is nil at the same time the error is nil?

Reply:
it should not, i forgot to continue where we get header error, now should be fixed